core/services/ocr2/plugins/threshold/plugin.go

```diff
@@ -0,0 +1,78 @@
+package threshold
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	libocr2 "github.com/smartcontractkit/libocr/offchainreporting2plus"
+
+	decryptionPlugin "github.com/smartcontractkit/tdh2/go/ocr2/decryptionplugin"
+	decryptionPluginConfig "github.com/smartcontractkit/tdh2/go/ocr2/decryptionplugin/config"
+	"github.com/smartcontractkit/tdh2/go/tdh2/tdh2easy"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type ThresholdServicesConfig struct {
+	DecryptionQueue    decryptionPlugin.DecryptionQueuingService
+	KeyshareWithPubKey []byte
+	ConfigParser       decryptionPluginConfig.ConfigParser
+}
+
+func NewThresholdService(sharedOracleArgs *libocr2.OCR2OracleArgs, conf *ThresholdServicesConfig) (job.ServiceCtx, error) {
+	publicKey, privKeyShare, err := UnmarshalKeys(conf.KeyshareWithPubKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal threshold key share with public key")
+	}
+
+	// The key generation tooling ensures that key IDs correspond to the oracle's index,
+	// therefore an identity mapping is used when creating the threshold reporting plugin.
+	oracleToKeyShare := make(map[commontypes.OracleID]int)
+	numNodes := 4
```

Comment:
We need a better approach here.  Do we think it is worth modifying the plugin directly to eliminate this map entirely in favor of using the indices from the `transmitters` array?
That would require future users of the plugin to also assign key shares in the same ordering as the `transmitters` array, however this seems to be the most straightforward approach.

Reply:
yes